pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -170,80 +154,93 @@ func createOrAddCustomAppsDomain(r *ReconcileCustomDomain, reqLogger logr.Logger
 	secretName := instance.Name
 
-	// create or update secret in openshift-ingress
+	// create secret in the openshift-ingress namespace
 	ingressSecret := &corev1.Secret{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{
 		Namespace: ingressNamespace,
 		Name:      secretName,
 	}, ingressSecret)
-	ingressSecret.Name = secretName
-	ingressSecret.Namespace = ingressNamespace
-	ingressSecret.Data = userSecret.Data
-	ingressSecret.Type = userSecret.Type
 	if err != nil {
-		err = r.client.Create(context.TODO(), ingressSecret)
-		if err != nil {
-			reqLogger.Error(err, "Error creating custom certificate secret")
-			return err
+		if errors.IsNotFound(err) {
+			ingressSecret.Name = secretName
+			ingressSecret.Namespace = ingressNamespace
+			ingressSecret.Data = userSecret.Data
+			ingressSecret.Type = userSecret.Type
+			err = r.client.Create(context.TODO(), ingressSecret)
+			if err != nil {
+				reqLogger.Error(err, "Error creating custom certificate secret")
+				return reconcile.Result{}, err
+			}
 		}
 	} else {
-		err = r.client.Update(context.TODO(), ingressSecret)
-		if err != nil {
-			reqLogger.Error(err, "Error updating custom certificate secret")
-			return err
-		}
+		reqLogger.Info(fmt.Sprintf("Certificate secret %s already exists in the %s namespace", secretName, ingressNamespace))
 	}
 
-	// get dnses.config.openshift.io/cluster for installed base domain
+	// get dnses.config.openshift.io/cluster for base domain
 	dnsConfig := &configv1.DNS{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{
-		Name: "cluster",
+		Name: dnsConfigName,
 	}, dnsConfig)
 	if err != nil {
-		reqLogger.Error(err, "Error getting dns.config/cluster")
-		return err
+		reqLogger.Error(err, fmt.Sprintf("Error getting dns.config/%s", dnsConfigName))
+		return reconcile.Result{}, err
 	}
 
 	// set the ingress domain to be a subdomain under the cluster's installed basedomain
 	// such that the record is added to the zone and external DNS can point to it
 	ingressDomain := fmt.Sprintf("%s.%s", instance.Name, dnsConfig.Spec.BaseDomain)
 	ingressName := instance.Name
 
-	// create or update ingresscontrollers.openshift.io/custom
+	// create new ingresscontrollers.openshift.io
 	customIngress := &operatorv1.IngressController{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{
 		Namespace: ingressOperatorNamespace,
 		Name:      ingressName,
 	}, customIngress)
-	customIngress.Name = ingressName
-	customIngress.Namespace = ingressOperatorNamespace
-	customIngress.Spec.Domain = ingressDomain
-	if customIngress.Spec.DefaultCertificate != nil {
-		customIngress.Spec.DefaultCertificate.Name = secretName
-	} else {
-		customIngress.Spec.DefaultCertificate = &corev1.LocalObjectReference{Name: secretName}
-	}
 	if err != nil {
-		err = r.client.Create(context.TODO(), customIngress)
-		if err != nil {
-			reqLogger.Error(err, "Error creating custom ingress controller")
-			return err
+		if errors.IsNotFound(err) {
```

Comment:
Same here

Reply:
Thanks, I've fixed it to return err in the else block.